zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/approval/ApproveTask.java

```diff
@@ -492,13 +492,20 @@ protected String filePathGenerator(String datasetId, Date publishDate, String Da
         String finalPath;
 
         if (nonTsDatasetWhitelistSet.contains(baseFilename)) {
-            finalPath = "ts-datasets/" + "other" + "/" + formattedDate;
+            finalPath = "ts-datasets/" + "other" + "/" + formattedDate + "/" + pickCorrectDatasetVersion(correctDatasetVersion, DatasetVersion);
         } else {
-            finalPath = "ts-datasets/" + baseFilename + "/" + (!correctDatasetVersion.equals("")? incrementDatasetVersionByOne(correctDatasetVersion) : DatasetVersion);
+            finalPath = "ts-datasets/" + baseFilename + "/" + pickCorrectDatasetVersion(correctDatasetVersion, DatasetVersion);
         }
         return finalPath;
     }
 
+    protected String pickCorrectDatasetVersion(String correctDatasetVersion, String datasetVersion) {
```

Comment:
The naming of this could be improved. 

The name of the method is currently `pickCorrectDatasetVersion` - and we're passing in two datasetVersions, one of which is labelled as correct.